### DISCUSSION TITLE
S8 inbound roaming and IPX connections?

### DISCUSSION THREAD

**Discussion by jpad-freedomfi** [magma/magma#7012] [Q&A]
Hi All, 

We are working on an inbound roaming setup for a customer. We have been working through getting connection to an IPX/GRX network. 

In the discussions, it has come out that traditionally the SGW has a globally routable IP address that is not advertised on the public internet. Instead it is advertised on the GRX network as the GTP-U endpoint for S8 traffic. With a distributed AGW architecture, this make scaling tricky if we get a lot of sites, we need a IPv4 address address to share with our GRX vendor for each site. 

So each site would have IPsec connection from private subnet at the site to GRX vendor IPsec aggregator. The source of this would be the whole site public IP from edge router. Then the SGW would have a 1:1 NAT for this other globally routable IP that would be advertised to GRX provider via BGP over the IPsec. The SGW IP is what would be used to route GTP-U traffic in the GRX network. 

Curious if others have heard of this design? And/or how others have handled GTP-U setup with Magma S8 inbound roaming?

**Comment by ekowtaylor** [comment]
cc @pshelar

**Reply by pshelar** [reply]
Inbound roaming user-plane networking uses host routing. so this design would work with current inbound roaming.

**Reply by jpad-freedomfi** [reply]
Thanks @pshelar but my question is can it be done without the need for a public IP for each AGW? That makes scaling hard.

**Reply by pshelar** [reply]
@jpad-freedomfi  yes, there is no hard requirement for public IP for S8 tunnel connectivity. S8 interface is configurable in MME config. you can assign non public IP to the interface to routes S8 packets.

**Reply by amarpad** [reply]
How does this work with a split GTP-C and U like in Magma?

**Reply by amarpad** [reply]
Also when you say the SGW has a NAT, can you help me understand. Is that the Magma SGW?

**Comment by jpad-freedomfi** [comment]
@amarpad see the diagram attached. 

When I say the SGW has a NAT, I mean the AGW is getting a private IP on SGi interface. (so yes that is magma SGW)

And you question about how it works with split GTP-C and U is what we are working through. The fact that SGW is distributed means in this model we need a public IP for the GTP-U endpoint (separate from site pub IP from ISP). This makes scaling a challenge. 
[S8 Inbound via IPX.pdf](https://github.com/magma/magma/files/6548503/S8.Inbound.via.IPX.pdf)
